src/scripts.js

```diff
@@ -150,10 +181,32 @@ function toggleFavorites(event) {
   })
 }
 
+function displayRedHearts(list) {
```

Comment:
Glad these functions resolved the issue that didn't persist the colors of the icons on different views.